misc/debug-adapter/modules/debug-adapter-core/src/main/java/org/ballerinalang/debugadapter/config/ClientLaunchConfigHolder.java

```diff
@@ -0,0 +1,154 @@
+import java.util.Optional;
+
+/**
+ * Client configuration holder implementation for launch-mode.
+ *
+ * @since 2.0.0
+ */
+public class ClientLaunchConfigHolder extends ClientConfigHolder {
+
+    private Boolean noDebug;
+    private String ballerinaHome;
+    private String ballerinaCommand;
+    private Boolean testDebug;
+    private Boolean networkLogsEnabled;
+    private Integer networkLogsPort;
+    private List<String> commandOptions;
+    private List<String> programArguments;
+    private Map<String, String> env;
+
+    private static final String ARG_BALLERINA_HOME = "ballerina.home";
+    private static final String ARG_BALLERINA_COMMAND = "ballerina.command";
+    private static final String ARG_TEST_DEBUG = "debugTests";
+    private static final String ARG_NETWORK_LOGS = "networkLogs";
+    private static final String ARG_NETWORK_LOGS_PORT = "networkLogsPort";
+    private static final String ARG_COMMAND_OPTIONS = "commandOptions";
+    private static final String ARG_SCRIPT_ARGUMENTS = "scriptArguments";
+    private static final String ARG_NO_DEBUG = "noDebug";
+    private static final String ARG_ENV = "env";
+
+    public ClientLaunchConfigHolder(Map<String, Object> args) {
+        super(args, ClientConfigKind.LAUNCH_CONFIG);
+    }
+
+    public String getBallerinaHome() throws ClientConfigurationException {
+        if (ballerinaHome == null) {
+            failIfMissing(ARG_BALLERINA_HOME);
+            ballerinaHome = clientRequestArgs.get(ARG_BALLERINA_HOME).toString();
+        }
+        return ballerinaHome;
+    }
+
+    public Optional<String> getBallerinaCommand() {
+        if (ballerinaCommand == null) {
+            if (!clientRequestArgs.containsKey(ARG_BALLERINA_COMMAND)) {
+                return Optional.empty();
+            }
+            ballerinaCommand = clientRequestArgs.get(ARG_BALLERINA_COMMAND).toString();
+        }
+        return Optional.ofNullable(ballerinaCommand);
+    }
+
+    public boolean isNoDebug() {
+        if (noDebug == null) {
+            Object noDebugObject = clientRequestArgs.get(ARG_NO_DEBUG);
+            if (noDebugObject instanceof Boolean) {
+                noDebug = (Boolean) noDebugObject;
+            } else if (noDebugObject instanceof String) {
+                noDebug = Boolean.parseBoolean((String) noDebugObject);
+            } else {
+                noDebug = false;
+            }
+        }
+        return noDebug;
+    }
+
+    public boolean isTestDebug() {
+        if (testDebug == null) {
+            Object testDebugObj = clientRequestArgs.get(ARG_TEST_DEBUG);
+            if (testDebugObj instanceof Boolean) {
+                testDebug = (Boolean) testDebugObj;
+            } else if (testDebugObj instanceof String) {
+                testDebug = Boolean.parseBoolean((String) testDebugObj);
+            } else {
+                testDebug = false;
+            }
+        }
+        return testDebug;
+    }
+
+    public boolean networkLogsEnabled() {
```

Comment:
How about changing the name to `isNetworkLogsEnabled` here?

Reply:
Renamed